src/helpers/figures/charts/runtime/chart_data_extractor.ts

```diff
@@ -0,0 +1,624 @@
+  }
+  if (definition.aggregated) {
+    ({ labels, dataSetsValues } = aggregateDataForLabels(labels, dataSetsValues));
+  }
+
+  const leftAxisFormat = getChartDatasetFormat(getters, dataSets, "left");
+  const rightAxisFormat = getChartDatasetFormat(getters, dataSets, "right");
+  const labelsFormat = getChartLabelFormat(getters, labelRange);
+  const axisFormats = { y: leftAxisFormat, y1: rightAxisFormat, x: labelsFormat };
+
+  const trendDatasets: ((number | null)[] | undefined)[] = [];
+  for (const index in dataSetsValues) {
+    let { data } = dataSetsValues[index];
+    if (definition.cumulative) {
+      let accumulator = 0;
+      data = data.map((value) => {
+        if (!isNaN(value)) {
+          accumulator += parseFloat(value);
+          return accumulator;
+        }
+        return value;
+      });
+      dataSetsValues[index] = { ...dataSetsValues[index], data };
+    }
+
+    const trend = definition.dataSets?.[index].trend;
+    if (!trend?.display) {
+      trendDatasets.push(undefined);
+      continue;
+    }
+
+    trendDatasets.push(getTrendDatasetForLineChart(trend, data, axisType, getters.getLocale()));
+  }
+
+  return {
+    dataSetsValues,
+    axisFormats,
+    labels,
+    locale: getters.getLocale(),
+    trendDatasets,
+    axisType,
+  };
+}
+
+export function getPieChartData(
+  definition: GenericDefinition<PieChartDefinition>,
+  dataSets: DataSet[],
+  labelRange: Range | undefined,
+  getters: Getters
+): ChartRuntimeGenerationArgs {
+  const labelValues = getChartLabelValues(getters, dataSets, labelRange);
+  let labels = labelValues.formattedValues;
+  let dataSetsValues = getChartDatasetValues(getters, dataSets);
+  if (
+    definition.dataSetsHaveTitle &&
+    dataSetsValues[0] &&
+    labels.length > dataSetsValues[0].data.length
+  ) {
+    labels.shift();
+  }
+
+  ({ labels, dataSetsValues } = filterEmptyDataPoints(labels, dataSetsValues));
+
+  if (definition.aggregated) {
+    ({ labels, dataSetsValues } = aggregateDataForLabels(labels, dataSetsValues));
+  }
+
+  ({ dataSetsValues, labels } = filterNegativeValues(labels, dataSetsValues));
+
+  const dataSetFormat = getChartDatasetFormat(getters, dataSets, "left");
+
+  return {
+    dataSetsValues,
+    axisFormats: { y: dataSetFormat },
+    labels,
+    locale: getters.getLocale(),
+  };
+}
+
+export function getTrendDatasetForBarChart(config: TrendConfiguration, data: any[]) {
+  const filteredValues: number[] = [];
+  const filteredLabels: number[] = [];
+  const labels: number[] = [];
+  for (let i = 0; i < data.length; i++) {
+    if (typeof data[i] === "number") {
+      filteredValues.push(data[i]);
+      filteredLabels.push(i + 1);
+    }
+    labels.push(i + 1);
+  }
+
+  const newLabels = range(0.5, labels.length + 0.55, 0.2);
+  const newValues = interpolateData(config, filteredValues, filteredLabels, newLabels);
+  return newValues.length ? newValues : undefined;
+}
+
+export function getTrendDatasetForLineChart(
```

Comment:
It's used in Odoo. It's not showing in the index.ts because it'n included in the 

```js
import * as CHART_RUNTIME_HELPERS from "./helpers/figures/charts/runtime";

```